it/v1/endpoints/CreateForeignPropertyControllerISpec.scala

```diff
@@ -0,0 +1,645 @@
+      "/foreignProperty/0/expenditure/premisesRunningCosts",
+      "/foreignProperty/0/expenditure/repairsAndMaintenance",
+      "/foreignProperty/0/expenditure/financialCosts",
+      "/foreignProperty/0/expenditure/professionalFees",
+      "/foreignProperty/0/expenditure/costsOfServices",
+      "/foreignProperty/0/expenditure/travelCosts",
+      "/foreignProperty/0/expenditure/residentialFinancialCost",
+      "/foreignProperty/0/expenditure/broughtFwdResidentialFinancialCost",
+      "/foreignProperty/0/expenditure/other",
+    ))
+  )
+
+  val allInvalidCountryCodeRequestError: MtdError = CountryCodeFormatError.copy(
+    message = "The provided Country code is invalid",
+    paths = Some(List(
+      "/foreignProperty/0/countryCode"
+    ))
+  )
+
+  val allRuleCountryCodeRequestError: MtdError = RuleCountryCodeError.copy(
+    message = "The country code is not a valid ISO 3166-1 alpha-3 country code",
+    paths = Some(List(
+      "/foreignProperty/0/countryCode"
+    ))
+  )
+
+  val RuleBothExpensesSuppliedRequestError: MtdError = RuleBothExpensesSuppliedError.copy(
+    message = "Both expenses and consolidatedExpenses can not be present at the same time",
+    paths = Some(List(
+      "/foreignFhlEea/expenditure"
+    ))
+  )
+
+
+  private trait Test {
+    val nino = "TC663795B"
+    val businessId = "XAIS12345678910"
+
+
+    def setupStubs(): StubMapping
+
+    def uri: String
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders((ACCEPT, "application/vnd.hmrc.1.0+json"))
+    }
+
+    val responseBody = Json.parse(
+      """
+        |{
+        |  "submissionId": "4557ecb5-fd32-48cc-81f5-e6acd1099f3c",
+        |  "links": [
+        |    {
+        |      "href":"/individuals/business/property/TC663795B/XAIS12345678910/period/4557ecb5-fd32-48cc-81f5-e6acd1099f3c",
+        |      "method":"GET",
+        |      "rel":"self"
+        |    }
+        |  ]
+        |}
+        |""".stripMargin)
+
+    def errorBody(code: String): String =
+      s"""
+         |      {
+         |        "code": "$code",
+         |        "reason": "des message"
+         |      }
+    """.stripMargin
+  }
+
+  "calling the create endpoint" should {
+
+    trait CreateTest extends Test {
+      def uri: String = s"/$nino/$businessId/period"
+
+      def desUri: String = s"/business/property/$nino/$businessId/period"
+
+      val desResponse = Json.parse(
+        """
+          |{
+          |  "submissionId": "4557ecb5-fd32-48cc-81f5-e6acd1099f3c",
+          |  "links": [
+          |    {
+          |      "href":"/individuals/business/property/TC663795B/XAIS12345678910/period/4557ecb5-fd32-48cc-81f5-e6acd1099f3c",
+          |      "method":"GET",
+          |      "rel":"self"
+          |    }
+          |  ]
+          |}
+          |""".stripMargin)
+    }
+
+    "return a 201 status" when {
+
+      "any valid request is made" in new CreateTest {
```

Comment:
Can you also include a valid consolidated test like you have with the controller tests?